runtime/commonMain/src/kotlinx/serialization/json/JsonExceptions.kt

```diff
@@ -44,3 +45,9 @@ public class JsonParsingException(position: Int, message: String) : JsonExceptio
  * ```
  */
 public class JsonElementTypeMismatchException(key: String, expected: String) : JsonException("Element $key is not a $expected")
+
+public class JsonMapInvalidKeyKind(keyDescriptor: SerialDescriptor) : JsonException(
```

Comment:
Please do not introduce new public exception type for a rare non-recoverable situation, which is caught and fixed by the developer in unit-tests. A verbose exception is enough

Reply:
Problem is that generic `JsonException` is a sealed (i.e. abstract) class. I think we need to revisit json exception hierarchy

Reply:
Let's discuss it tomorrow then